contracts/tests/src/contract.rs

```diff
@@ -0,0 +1,69 @@
+use cosmwasm_std::{
+    entry_point, to_binary, Deps, DepsMut, Empty, Env, MessageInfo, QueryResponse, Response,
+    StdError, StdResult,
+};
+
+use crate::errors::HackError;
+use crate::msg::{ExecuteMsg, QueryMsg};
+
+#[entry_point]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    _msg: Empty,
+) -> Result<Response, HackError> {
+    deps.api.debug("here we go 🚀");
+
+    // This adds some unrelated event attribute for testing purposes
+    Ok(Response::new().add_attribute("Let the", "hacking begin"))
+}
+
+#[entry_point]
+pub fn execute(
+    _deps: DepsMut,
+    env: Env,
+    _info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, HackError> {
+    use ExecuteMsg::*;
+
+    match msg {
+        Argon2 {
+            mem_cost,
+            time_cost,
+        } => do_argon2(mem_cost, time_cost),
+        MirrorEnv {} => Ok(Response::new().set_data(to_binary(&env)?)),
+    }
+}
+
+fn do_argon2(mem_cost: u32, time_cost: u32) -> Result<Response, HackError> {
```

Comment:
Renaming this to `fn execute_argon2` would be more consistent with the `fn query_*` style we usually use for queries.

Reply:
Sounds great. Let's do that for all new contracts.

Reply:
is `fn execute` and `mod execute` in the same module not a problem?

Reply:
It isn't. I think those are separate namespaces